src/js/store/middleware/feature/form.js

```diff
@@ -226,6 +265,46 @@ const handleClear = (store, next) => {
   ])
 }
 
+const handlePersistInLocalStorage = (store) => {
+  const state = store.getState()
+  const key = localStorageKey(state)
+  const storageState = {
+    schema: state.form.schema,
+    fields: state.form.fields,
+  }
+  window.localStorage.setItem(key, JSON.stringify(storageState))
+}
+
+const handleClearLocalStorage = (store) => {
+  const state = store.getState()
+  const key = localStorageKey(state)
+  window.localStorage.removeItem(key)
+}
+
+const handleAcceptLocalStorage = (store, next) => {
+  const state = store.getState()
+  const key = localStorageKey(state)
+  let storageState = { fields: {} }
+  try {
+    storageState = JSON.parse(window.localStorage.getItem(key))
+  } catch(error) {
+    console.error(error)
+  }
+  Object.entries(storageState.fields).forEach(([fieldId, value]) => {
+    store.dispatch(inputField({ fieldId, value }))
+  })
+  next(setAskingToRestore({ state: false, feature: FORM }))
+}
+
+const handleRejectLocalStorage = (store, next) => {
+  const state = store.getState()
+  state.form.schema.columns.forEach(col => {
+    store.dispatch(setField({ fieldId: col.id, value: parseDefault(col.config.default, col.type) }))
+  })
+  store.dispatch(clearLocalStorage())
+  next(setAskingToRestore({ state: false, feature: FORM }))
+}
```

Comment:
When we reject local storage (either because a user said "no" to the prompt or because the schema was invalid or because there was nothing in local storage), we clear the contents of local storage for this form, set the `askingToRestore` flag to false, and set all of the fields to their default values. Setting fields to their default values is what we would do on form load before this change, so the fact that we automatically "reject" local storage when it's empty preserves the old behavior when no local data is found.